tests/unit/modules/test_tessen.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import pytest
+
+from ansible_collections.sensu.sensu_go.plugins.module_utils import (
+    errors, response, utils
+)
+from ansible_collections.sensu.sensu_go.plugins.modules import tessen
+
+from .common.utils import (
+    AnsibleExitJson, AnsibleFailJson, ModuleTestCase, set_module_args,
+)
+
+
+class TestSync:
+    def test_remote_none(self, mocker):
+        client = mocker.Mock()
+        client.get.side_effect = (
+            response.Response(404, ''),
+            response.Response(200, '{"opt_out": "true"}'),
+        )
+        client.put.return_value = response.Response(201, "")
```

Comment:
Is this a valid scenario? It feels a bit strange for API that deals with a static resource to return 404. If the API docs do not list the 404 as a valid return status, I would remove this test and add warn users about the failure. This probably means that we will need to use the Sensu client directly instead of relying on the helpers from `utils` module.